terraform/main.tf

```diff
@@ -0,0 +1,155 @@
+resource "aws_s3_bucket" "batugo-autoscribe" {
+  bucket = "batugo-autoscribe"
+
+  tags = {
+    Owner = element(split("/", data.aws_caller_identity.current.arn), 1)
+  }
+}
+
+resource "aws_s3_bucket_ownership_controls" "batugo-autoscribe_ownership_controls" {
+  bucket = aws_s3_bucket.batugo-autoscribe.id
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
+resource "aws_s3_bucket_acl" "batugo-autoscribe_acl" {
+  depends_on = [aws_s3_bucket_ownership_controls.batugo-autoscribe_ownership_controls]
+
+  bucket = aws_s3_bucket.batugo-autoscribe.id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "batugo-autoscribe_expiration" {
+  bucket = aws_s3_bucket.batugo-autoscribe.id
+
+  rule {
+    id      = "compliance-retention-policy"
+    status  = "Enabled"
+
+    expiration {
+	  days = 100
+    }
+  }
+}
+
+resource "aws_ecr_repository" "abatugo_autoscribe" {
+  name                 = "abatugo_autoscribe"
+  image_tag_mutability = "MUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "aws_iam_role" "abatugo_autoscribe_ecs_task_execution_role" {
+  name = "abatugo_autoscribe_execution_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action    = "sts:AssumeRole"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+        Effect    = "Allow"
+        Sid       = ""
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "abatugo_autoscribe_ecs_task_execution_role_policy" {
+  role       = aws_iam_role.abatugo_autoscribe_ecs_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "abatugo_autoscribe_ecs_task_role" {
+  name = "abatugo_autoscribe_task_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action    = "sts:AssumeRole"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+        Effect    = "Allow"
+        Sid       = ""
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "abatugo_autoscribe_ecs_task_role_policy_attachment" {
+  role       = aws_iam_role.abatugo_autoscribe_ecs_task_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+}
+
+resource "aws_cloudwatch_log_group" "abatugo_autoscribe_task_log" {
+  name = "/ecs/abatugo_autoscribe_task_log"
+
+  tags = {
+    Environment = "production"
+    Application = "serviceA"
+  }
```

Comment:
I don't think these tags are strictly necessary unless you're using them for something else downstream